src/pages/plugins/plugins.js

```diff
@@ -66,48 +66,108 @@ export default function PluginsInclude(updates) {
+      nextPage: 1,
+      buffer: [],
+      hasMoreSource: true,
+    };
+
+    currentFilter = normalizedFilter;
+    currentPage = 1;
+    hasMore = true;
+    isLoading = false;
+    plugins.all = [];
+
+    if (currSection !== "all") {
+      render("all");
+    } else {
+      $list.all.replaceChildren();
+    }
+
+    const filterMessage = (
+      <div className="filter-message">
+        Filtered by <strong>{normalizedFilter.displayLabel}</strong>
+        <span className="icon clearclose" data-action="clear-filter" onclick={clearFilter}></span>
+      </div>
+    );
+
+    $list.all.append(filterMessage);
+    $list.all.setAttribute("empty-msg", strings["loading..."]);
+    await getFilteredPlugins(currentFilter, true);
+  }
+
+  function clearFilter() {
+    currentFilter = null;
+    currentPage = 1;
+    hasMore = true;
+    isLoading = false;
+    plugins.all = [];
+    $list.all.replaceChildren();
+    $list.all.setAttribute("empty-msg", strings["loading..."]);
+    getAllPlugins();
+  }
+
   Contextmenu({
     toggler: $filter,
     top: "8px",
     right: "16px",
     items: [
-      [strings.top_rated, "top_rated"],
-      [strings.newly_added, "newest"],
-      [strings.most_downloaded, "downloads"],
+      [strings.top_rated, "orderBy:top_rated"],
+      [strings.newly_added, "orderBy:newest"],
+      [strings.most_downloaded, "orderBy:downloads"],
+      [verifiedLabel, "attribute:verified"],
+      [authorLabel, "attribute:author"],
+      [keywordsLabel, "attribute:keywords"],
     ],
-    onselect(item) {
-      const filterNames = {
-        top_rated: strings.top_rated,
-        newest: strings.newly_added,
-        downloads: strings.most_downloaded,
+    async onselect(action) {
+      const option = filterOptions[action];
+      if (!option) return;
+
+      const filterState = {
+        type: option.type,
+        value: option.value,
+        baseLabel: option.baseLabel,
+        displayLabel: option.baseLabel,
       };
-      const filterName = filterNames[item];
-      currentFilter = item;
-      currentPage = 1;
-      hasMore = true;
-      isLoading = false;
-      plugins.all = []; // Reset the all plugins array
-      render("all");
-      $list.all.replaceChildren();
-      $list.all.append(
-        <div className="filter-message">
-          Filtered by <strong>{filterName}</strong>
-          <span
-            className="icon clearclose"
-            data-action="clear-filter"
-            onclick={() => {
-              currentFilter = null;
-              currentPage = 1;
-              hasMore = true;
-              isLoading = false;
-              plugins.all = []; // Reset the all plugins array
-              $list.all.replaceChildren();
-              getAllPlugins();
-            }}
-          ></span>
-        </div>,
-      );
-      getFilteredPlugins(item);
+
+      if (option.type === "author") {
+        const authorName = (await prompt("Enter author name", "", "text"))?.trim();
```

Comment:
Internationalisation is missing for `Enter author name`

```suggestion
        const authorName = (await prompt(strings["Enter author name"] || "Enter author name", "", "text"))?.trim();
```

Reply:
That's not necessary